### PR TITLE
[server] Allow locale and group separator override

### PR DESCRIPTION
**Pull request by elpaso**
New env vars:

QGIS_SERVER_OVERRIDE_SYSTEM_LOCALE (default '')
QGIS_SERVER_SHOW_GROUP_SEPARATOR (default false)

Note: default behavior is now to not show thousand separator.
Note on backport: I'd like to backport the override vars too, but if someone disagree I will backport just the default false behavior, because that's a bug.

Funded by: Kanton Zug

### CODE REVIEW

File: src/server/qgsserversettings.h [hunk 1]
```diff
@@ -158,16 +160,28 @@ class SERVER_EXPORT QgsServerSettings
...
      * Returns the cache directory.
-      * \returns the directory.
-      */
+     * \returns the directory.
+     */
     QString cacheDirectory() const;
 
+    /**
+     * Overrides system locale
+     * \returns the optional override for system locale.
```
Comment (pblottiere) [inline src/server/qgsserversettings.h]: \since

File: src/server/qgsserversettings.h [hunk 2]
```diff
@@ -158,16 +160,28 @@ class SERVER_EXPORT QgsServerSettings
...
+     */
+    QString overrideSystemLocale() const;
+
+    /**
+     * Show group (thousand) separator
+     * \returns if group separator must be shown, default to FALSE.
```
Comment (pblottiere) [inline src/server/qgsserversettings.h]: \since

**Comment by pblottiere** [comment]
It looks good to me, thanks @elpaso!

**Review by pblottiere — APPROVED** [review]

**Comment by jef-n** [comment]
Fixes #21687

**Pull request merged**